### DISCUSSION TITLE
WT32_ETH01. Failed to flashing

### DISCUSSION THREAD

**Discussion by superwebmx** [forkineye/ESPixelStick#621] [Q&A] [answered]
Hello, huge thanks to @forkineye and the contributors 4.x. Sorry for the length of the query.

I'm trying to flash a WT32-ETH01 v.1.4. board but I can't finish it.

1. I am using the ESPixelStick v4 beta4 version with the newly updated ESPSFlashTool.jar.

2. I have checked the wiring and everything is Ok. The WT32-ETH01 is ready to program.

3. When starting the flashing, the first stage passes (or so the log indicates)...
   -= Building Filesystem Image - Done =-

4. But in the second stage (erasing) it shows errors

```
Serial port COM4
Connecting........_____....._____..._____.._____....._____....._____....._____
Traceback (most recent call last):
   File "E:\wLED\ESPixelStick-clone\ESPixelStick_Firmware-4.0-ci4000828742\bin\upload.py", line 19, in <module>
     esptool.main(sys.argv)
   File "E:/wLED/ESPixelStick-clone/ESPixelStick_Firmware-4.0-ci4000828742/bin/esptool\esptool.py", line 2890, in main
     esp.connect(args.before)
   File "E:/wLED/ESPixelStick-clone/ESPixelStick_Firmware-4.0-ci4000828742/bin/esptool\esptool.py", line 483, in connect
     raise FatalError('Failed to connect to %s: %s' % (self.CHIP_NAME, last_error))
esptool.FatalError: Failed to connect to ESP32: Timed out waiting for packet header_
```

5. Uploading firmware also has errors
```
Serial port COM4
Connecting........_____....._____..._____.._____....._____....._____....._____
Traceback (most recent call last):
   File "E:\wLED\ESPixelStick-clone\ESPixelStick_Firmware-4.0-ci4000828742\bin\upload.py", line 19, in <module>
     esptool.main(sys.argv)
   File "E:/wLED/ESPixelStick-clone/ESPixelStick_Firmware-4.0-ci4000828742/bin/esptool\esptool.py", line 2890, in main
     esp.connect(args.before)
   File "E:/wLED/ESPixelStick-clone/ESPixelStick_Firmware-4.0-ci4000828742/bin/esptool\esptool.py", line 483, in connect
     raise FatalError('Failed to connect to %s: %s' % (self.CHIP_NAME, last_error))
esptool.FatalError: Failed to connect to ESP32: Timed out waiting for packet header
Command: bin/win32/python3/python bin/upload.py --chip esp32 --baud 460800 --port COM4 --before default_reset --after hard_reset write_flash -z 0x1000 firmware/esp32/esp32_wt32eth01-bootloader.bin 0x8000 firmware/ esp32/esp32_wt32eth01-partitions.bin 0xe000 firmware/esp32/boot_app0.bin 0x10000 firmware/esp32/esp32_wt32eth01-app.bin 0x3B0000 firmware/filesystem.bin - Done
```

6. The wiring and connection is correct. Without changing anything, I try to flash another system image ([wLed v0.14.0](https://github.com/Aircoookie/WLED/releases) for example) with their flashing tool ([ESPHome-Flasher-1.4.0-Windows-x64.exe](https://github.com/esphome/esphome-flasher/releases)) and the process is done without problems, and the system boot and works correctly.

7. As a test, with the tool ESPHome-Flasher-1.4.0-Windows-x64.exe I flash using the image [esp32_wt32eth01-app.bin](https://github.com/forkineye/ESPixelStick/releases/tag/v4.0-beta4) and it is upload correctly. The WT32 board boots fine but since it does not have the file system, the web interface indicates "page not found"

**Will there be a problem with the tool to flash this board?**
I have flashed with the same tool and version two other boards (ESP32 Quinled Octo and Olimex ESP32) without problem.

**Is there a way to use another flashing tool to correctly upload the bin files?**
In my case esp32_wt32eth01-app.bin, esp32_wt32eth01-bootloader.bin, esp32_wt32eth01-partitions.bin

**Comment by superwebmx** [comment, marked as answer]
Hi, apparently the problem is entering programming mode up to three times, despite carefully manipulating the programming pin I have not succeeded. :(
I don't understand why other flashing tools can recognize the programming status and this one doesn't.
I am finally doing the flashing compiling with PlatformIO and everything is going well.
Greetings...

:)

**Comment by MartinMueller2003** [comment]
I had a similar issue in https://github.com/forkineye/ESPixelStick/discussions/627 and posted a procedure. The problem is that for V4 flashing is a three iteration process and the WT device does not support MCU reset functionality which means you need to reset the device by hand at the right time.
